### ISSUE TITLE
Samsung Galaxy A3 (2015) [SM-A300FU] UnicodeDecodeError

### ISSUE THREAD

**Issue by inkstray** [bkerler/sboot_dump#6]
I cant dump anything from my Samsung Galaxy A3 (2015)
I get the following error:
```
Traceback (most recent call last):
  File "/home/tryzen/Code/sboot_dump/./samupload.py", line 716, in <module>
    main()
  File "/home/tryzen/Code/sboot_dump/./samupload.py", line 670, in main
    mode, devicename, probetable = suc.probe()
  File "/home/tryzen/Code/sboot_dump/./samupload.py", line 557, in probe
    return self.get_probe_table(data)
  File "/home/tryzen/Code/sboot_dump/./samupload.py", line 574, in get_probe_table
    pt = partitiontable(data.read(size), mode)
  File "/home/tryzen/Code/sboot_dump/./samupload.py", line 517, in __init__
    self.pname = sh.bytes(16).rstrip(b"\x00").decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe5 in position 11: invalid continuation byte
```

**Comment by timschneeb** [comment]
I'm experiencing a similar error with the Samsung Galaxy S20+ (SM-G985F):
```
$ python samupload.py partition 36

SUC - Samsung Upload Client v1.2 (c) B. Kerler 2018-2022, Email: info @ revskills.de
  CONFIGURATION 1: 50 mA ===================================
   bLength              :    0x9 (9 bytes)
   bDescriptorType      :    0x2 Configuration
   wTotalLength         :   0x43 (67 bytes)
   bNumInterfaces       :    0x2
   bConfigurationValue  :    0x1
   iConfiguration       :    0x0
   bmAttributes         :   0xc0 Self Powered
   bMaxPower            :   0x19 (50 mA)
    INTERFACE 0: CDC Communication =========================
     bLength            :    0x9 (9 bytes)
     bDescriptorType    :    0x4 Interface
     bInterfaceNumber   :    0x0
     bAlternateSetting  :    0x0
     bNumEndpoints      :    0x1
     bInterfaceClass    :    0x2 CDC Communication
     bInterfaceSubClass :    0x2
     bInterfaceProtocol :    0x1
     iInterface         :    0x0
      ENDPOINT 0x83: Interrupt IN ==========================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :   0x83 IN
       bmAttributes     :    0x3 Interrupt
       wMaxPacketSize   :   0x10 (16 bytes)
       bInterval        :    0x9
    INTERFACE 1: CDC Data ==================================
     bLength            :    0x9 (9 bytes)
     bDescriptorType    :    0x4 Interface
     bInterfaceNumber   :    0x1
     bAlternateSetting  :    0x0
     bNumEndpoints      :    0x2
     bInterfaceClass    :    0xa CDC Data
     bInterfaceSubClass :    0x0
     bInterfaceProtocol :    0x0
     iInterface         :    0x1 SAMSUNG
      ENDPOINT 0x81: Bulk IN ===============================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :   0x81 IN
       bmAttributes     :    0x2 Bulk
       wMaxPacketSize   :  0x200 (512 bytes)
       bInterval        :    0x0
      ENDPOINT 0x2: Bulk OUT ===============================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :    0x2 OUT
       bmAttributes     :    0x2 Bulk
       wMaxPacketSize   :  0x200 (512 bytes)
       bInterval        :    0x0
No kernel driver supported: Operation not supported or unimplemented on this platform
No kernel driver supported: Operation not supported or unimplemented on this platform

Downloading ....
-----------------
Progress: |--------------------------------------------------| 0.1% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x800 of 0x200000, 12s Progress: |--------------------------------------------------| 0.2% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x1000 of 0x200000, 19sProgress: |--------------------------------------------------| 0.3% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x1800 of 0x200000, 21sProgress: |--------------------------------------------------| 0.4% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x2000 of 0x200000, 22sProgress: |--------------------------------------------------| 0.5% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x2800 of 0x200000, 23sProgress: |--------------------------------------------------| 0.6% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x3000 of 0x200000, 23sProgress: |--------------------------------------------------| 0.7% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x3800 of 0x200000, 24sProgress: |--------------------------------------------------| 0.8% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x4000 of 0x200000, 24sProgress: |--------------------------------------------------| 0.9% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x4800 of 0x200000, 24sProgress: |--------------------------------------------------| 1.0% File: "ap_9c0000000_9ffffffff.lst" (Sector 0x5000 of 0x200000, 24s left) 41.25 MB/sTraceback (most recent call last):
  File "D:\sboot_dump\samupload.py", line 716, in <module>
    main()
  File "D:\sboot_dump\samupload.py", line 699, in main
    suc.download(probetable[area])
  File "D:\sboot_dump\samupload.py", line 642, in download
    if self.download_area(wfilename,area.pstart,area.pend):
  File "D:\sboot_dump\samupload.py", line 621, in download_area
    self.progress.show_progress('File: \"%s\"' % filename, pos, rend - rstart+1, True)
  File "D:\sboot_dump\samupload.py", line 412, in show_progress
    print_progress(prog, 100, prefix='Progress:',
  File "D:\sboot_dump\samupload.py", line 445, in print_progress
    sys.stdout.write('\r%s |%s| %s%s %s' % (prefix, bar, percents, '%', suffix))
  File "C:\Users\tim26\AppData\Local\Programs\Python\Python310\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2588' in position 12: character maps to <undefined>
```

**Comment by timschneeb** [comment]
In my case, the UnicodeDecodeError was caused by the '█' character used to draw the progress bar. I fixed it by replacing it with a '=' in the code instead.

**Comment by bkerler** [comment]
Yeah, windows often can cause issues with encoding. I'll have a look how it can be fixed appropriately

**Comment by bkerler** [comment]
Solved with commit 560e7fd3b1d41f4d10fb112ae3a546fdadf1a939
